File: server/src/notifications/notificationTypes.ts
import type { Session } from '../sync/syncEngine'

export type NotificationChannel = {
    sendReady: (session: Session) => Promise<void>
    sendPermissionRequest: (session: Session) => Promise<void>
    sendQuestion?: (session: Session, questionSummary: string) => Promise<void>
    sendError?: (session: Session, errorMessage: string) => Promise<void>
}

export type NotificationHubOptions = {
    readyCooldownMs?: number
    permissionDebounceMs?: number
    questionDebounceMs?: number
}

File: hub_go/internal/notifications/notification_types.go
package notifications

import "hub_go/internal/store"

type NotificationChannel interface {
	SendReady(session *store.Session) error
	SendPermissionRequest(session *store.Session) error
}

type NotificationHubOptions struct {
	ReadyCooldownMs      int64
	PermissionDebounceMs int64
}
